File: ThematicMapCreator.Api/Contracts/LayerStyleOptions/ChartDiagramStyleOptions.cs
using System;
using System.Collections.Generic;

namespace ThematicMapCreator.Api.Contracts.LayerStyleOptions
{
    public class ChartDiagramStyleOptions : ILayerStyleOptions
    {
        public string Color { get; set; }
        public string FillColor { get; set; }
        public Dictionary<string, string> PropertyNameColors { get; set; }
        public double Size { get; set; }
        public LayerStyle Style { get; } = LayerStyle.ChartDiagram;
    }
}

File: Sources/ThematicMapCreator.Client/src/app/models/layer-style-options/chart-diagram-style-options.ts
import { LayerStyleOptions } from './layer-style-options';
import { LayerStyle } from './layer-style.enum';

export class ChartDiagramStyleOptions implements LayerStyleOptions {
    style: LayerStyle.ChartDiagram;
    size = 8;
    color = '#000000';
    fillColor = '#000000';
    propertyNameColors: { [propertyName: string]: string } = {};
}
